core/modules/css/testKvInterfaceImplMySql.cc

```diff
@@ -0,0 +1,200 @@
+  * @brief Unit test for the MySql implementation of the Common State System Interface.
+  *
+  * @Author Nathan Pease, SLAC
+  */
+
+// System headers
+#include <iostream>
+#include <string>
+#include <unistd.h>
+
+// Boost headers
+#include <boost/algorithm/string/replace.hpp>
+
+// Boost unit test header
+#define BOOST_TEST_MODULE Css_1
+#include "boost/test/included/unit_test.hpp"
+
+// Qserv headers
+#include "css/CssError.h"
+#include "css/KvInterfaceImplMySql.h"
+#include "mysql/MySqlConfig.h"
+#include "sql/SqlConnection.h"
+#include "sql/SqlErrorObject.h"
+
+using lsst::qserv::css::KvInterfaceImplMySql;
+using lsst::qserv::mysql::MySqlConfig;
+using lsst::qserv::sql::SqlConnection;
+using lsst::qserv::sql::SqlErrorObject;
+
+namespace {
+
+// todo this is copied directly from testQMeta.cc. Perhaps it could be made generic and put in a shared location?
+struct TestDBGuard {
+    TestDBGuard() {
+        sqlConfig.hostname = "";
+        sqlConfig.port = 0;
+        sqlConfig.username = "root";
+        sqlConfig.password = getpass("Enter mysql root password: ");
+        std::cout << "Enter mysql socket: ";
+        std::cin >> sqlConfig.socket;
+
+        //create a db name with username & random number
+        sqlConfig.dbName = "testCSSZ012sdrt"; // todo: generate something nicely randomized
+
+        std::ifstream schemaFile("admin/templates/configuration/tmp/configure/sql/CssData.sql");
+
+        // read whole file into buffer
+        std::string buffer;
+        std::getline(schemaFile, buffer, '\0');
+
+        // replace production schema name with test schema
+        boost::replace_all(buffer, "qservCssData", sqlConfig.dbName);
+
+        // need config without database name
+        MySqlConfig sqlConfigLocal = sqlConfig;
+        sqlConfigLocal.dbName = "";
+        SqlConnection sqlConn(sqlConfigLocal);
+
+        SqlErrorObject errObj;
+        sqlConn.runQuery(buffer, errObj);
+        if (errObj.isSet()) {
+            throw lsst::qserv::css::CssError(errObj);
+        }
+    }
+
+    ~TestDBGuard() {
+        SqlConnection sqlConn(sqlConfig);
+        SqlErrorObject errObj;
+        sqlConn.dropDb(sqlConfig.dbName, errObj);
+    }
+
+    MySqlConfig sqlConfig;
+};
+
+} // namespace
+
+struct PerTestFixture
+{
+    PerTestFixture() {
+        kvInterface = std::make_shared<KvInterfaceImplMySql>(testDB.sqlConfig);
+        sqlConn = std::make_shared<SqlConnection>(testDB.sqlConfig);
+    }
+
+    static TestDBGuard testDB;
+    std::shared_ptr<SqlConnection> sqlConn;
+    std::shared_ptr<KvInterfaceImplMySql> kvInterface;
+};
+
+TestDBGuard PerTestFixture::testDB;
+
+
+BOOST_FIXTURE_TEST_SUITE(SqlKVInterfaceConnectionTestSuite, PerTestFixture)
+
+
+BOOST_AUTO_TEST_CASE(CreateAndGetKV) {
+    kvInterface->create("/foo/testKey", "testValue");
+    BOOST_CHECK_EQUAL(kvInterface->get("/foo/testKey"), "testValue");
+}
+
+
+BOOST_AUTO_TEST_CASE(CreateWithBackslash) {
+    kvInterface->create("\\foo\\testKey", "testValue");
```

Comment:
I don’t know, but I thought I would check that it works at least as well as that. I can nuke it entirely though. Any preference? Should I sanitize against backslash characters in key strings?

> On Jul 30, 2015, at 11:18 AM, Andy <notifications@github.com> wrote:
> 
> In core/modules/css/testKvInterfaceImplMySql.cc <https://github.com/lsst/qserv/pull/111#discussion_r35902155>:
> 
> > +};
> > +
> > +TestDBGuard PerTestFixture::testDB;
> > +
> > +
> > +BOOST_FIXTURE_TEST_SUITE(SqlKVInterfaceConnectionTestSuite, PerTestFixture)
> > +
> > +
> > +BOOST_AUTO_TEST_CASE(CreateAndGetKV) {
> > +    kvInterface->create("/foo/testKey", "testValue");
> > +    BOOST_CHECK_EQUAL(kvInterface->get("/foo/testKey"), "testValue");
> > +}
> > +
> > +
> > +BOOST_AUTO_TEST_CASE(CreateWithBackslash) {
> > +    kvInterface->create("\\foo\\testKey", "testValue");
> Hmm, are we required to support backslashes?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/lsst/qserv/pull/111/files#r35902155>.
>